packages/eslint-plugin/tests/rules/consistent-type-assertions.test.ts

```diff
@@ -790,5 +790,126 @@ ruleTester.run('consistent-type-assertions', rule, {
         },
       ],
     },
+    {
+      code: 'const a = <any>(b, c);',
+      output: `const a = (b, c) as any;`,
+      options: [
+        {
+          assertionStyle: 'as',
+        },
+      ],
+      errors: [
+        {
+          messageId: 'as',
+          line: 1,
+        },
+      ],
+    },
+    {
+      code: 'const f = <any>(() => {});',
+      output: 'const f = (() => {}) as any;',
+      options: [
+        {
+          assertionStyle: 'as',
+        },
+      ],
+      errors: [
+        {
+          messageId: 'as',
+          line: 1,
+        },
+      ],
+    },
+    {
+      code: 'const f = <any>function () {};',
+      output: 'const f = function () {} as any;',
+      options: [
+        {
+          assertionStyle: 'as',
+        },
+      ],
+      errors: [
+        {
+          messageId: 'as',
+          line: 1,
+        },
+      ],
+    },
+    {
+      code: 'const f = <any>(async () => {});',
+      output: 'const f = (async () => {}) as any;',
+      options: [
+        {
+          assertionStyle: 'as',
+        },
+      ],
+      errors: [
+        {
+          messageId: 'as',
+          line: 1,
+        },
+      ],
+    },
+    {
+      // prettier wants to remove the parens around the yield expression,
```

Comment:
I _think_ this is a prettier bug but tbh I'm not 100% certain. Filed https://github.com/prettier/prettier/issues/16192, we'll see what they say.

Reply:
was able to get this bug fixed. This will play nicely with https://github.com/typescript-eslint/typescript-eslint/issues/6128

Reply:
Nice!! Love to see an upstream bug discovered as part of our testing 😄